hedera-evm/src/main/java/com/hedera/evm/execution/EvmTxProcessor.java

```diff
@@ -0,0 +1,476 @@
+    final var precompileContractRegistry = new PrecompileContractRegistry();
+    MainnetPrecompiledContracts.populateForIstanbul(
+        precompileContractRegistry, this.gasCalculator);
+
+    this.messageCallProcessor =
+        new HederaMessageCallProcessor(
+            evm, precompileContractRegistry, precompiledContractMap);
+    this.contractCreationProcessor =
+        new ContractCreationProcessor(gasCalculator, evm, true, VALIDATION_RULES, 1);
+    this.blockMetaSource = blockMetaSource;
+  }
+
+
+  /**
+   * Executes the {@link MessageFrame} of the EVM transaction. Returns the result as {@link
+   * TransactionProcessingResult}
+   *
+   * @param sender
+   * 		The origin {@link Account} that initiates the transaction
+   * @param receiver
+   * 		the priority form of the receiving {@link Address} (i.e., EIP-1014 if present); or the newly created address
+   * @param gasPrice
+   * 		GasPrice to use for gas calculations
+   * @param gasLimit
+   * 		Externally provided gas limit
+   * @param value
+   * 		Evm transaction value (HBars)
+   * @param payload
+   * 		Transaction payload. For Create transactions, the bytecode + constructor arguments
+   * @param contractCreation
+   * 		Whether or not this is a contract creation transaction
+   * @param consensusTime
+   * 		Current consensus time
+   * @param isStatic
+   * 		Whether the execution is static
+   * @param mirrorReceiver
+   * 		the mirror form of the receiving {@link Address}; or the newly created address
+   * @return the result of the EVM execution returned as {@link TransactionProcessingResult}
+   */
+  public TransactionProcessingResult execute(
+      final Account sender,
+      final Address receiver,
+      final long gasPrice,
+      final long gasLimit,
+      final long value,
+      final Bytes payload,
+      final boolean contractCreation,
+      final Instant consensusTime,
+      final boolean isStatic,
+      final Address mirrorReceiver,
+      final BigInteger userOfferedGasPrice,
+      final long maxGasAllowanceInTinybars,
+      final Account relayer
+  ) {
+    ensureLatestChainId();
+    final Wei gasCost = Wei.of(Math.multiplyExact(gasLimit, gasPrice));
+    final Wei upfrontCost = gasCost.add(value);
+    final long intrinsicGas =
+        gasCalculator.transactionIntrinsicGasCost(Bytes.EMPTY, contractCreation);
+
+    final var updater = worldState.updater();
+    final var chargingResult =
+        chargeForGas(
+            gasCost,
+            upfrontCost,
+            value,
+            maxGasAllowanceInTinybars,
+            intrinsicGas,
+            gasPrice,
+            gasLimit,
+            isStatic,
+            userOfferedGasPrice,
+            sender.getId().asEvmAddress(),
+            relayer == null ? null : relayer.getId().asEvmAddress(),
+            updater);
+
+    final var coinbase = Id.fromGrpcAccount(dynamicProperties.fundingAccount()).asEvmAddress();
```

Comment:
You can replace `dynamicProperties` with `configurationProperties` and add all needed methods in the interface.